samples/rest/bing_auto_suggest_v7.py

```diff
@@ -0,0 +1,71 @@
+"""Python example showcasing the usage of Bing Auto Suggest API using REST calls"""
+
+import json
+from pprint import pprint
+
+import dotenv
+import requests
+
+
+def auto_suggest_basic(
+    subscription_key,
+    auth_header_name="Ocp-Apim-Subscription-Key",
+    query="sail",
+    mkt="en-us",
+):
+    """Bing Auto Suggest Basic REST call
+
+    This sample uses the Bing Autosuggest API to provide a list of suggested search query strings
+    Documentation: https://docs.microsoft.com/en-us/bing/search-apis/bing-autosuggest/overview
+
+    Args:
+        subscription_key (str): Azure subscription key of Bing Auto Suggest service
+        auth_header_name (str): Name of the authorization header
+        query (str): Query to get suggestions for
+        mkt (str): Market to search in
+    """
+
+    # Construct the request
+    endpoint = "https://api.bing.microsoft.com/v7.0/suggestions"
+    params = {"q": query, "mkt": mkt}
+    headers = {auth_header_name: subscription_key}
+
+    # Call the API
+    try:
+        response = requests.get(endpoint, headers=headers, params=params, timeout=10)
+        response.raise_for_status()
+        return response
+    except Exception as ex:
+        raise Exception(f"Encountered exception: {ex}") from ex
```

Comment:
\*My comment applies to all the files\*
I see no need for a try and catch here since we're not doing anything except re-throwing the error with extra `Encountered exception: ` prefix. We're also doing the same in the main function which means that you'll have the `Encountered exception: ` message duplicated. Not sure if you tried to run the code but it produces the following output
![image](https://github.com/microsoft/bing-search-sdk-for-python/assets/36763164/e53a5e42-090a-4e25-94d6-893f29eaf17b)

I, personally, think that not handling the exception would result in a better error reporting. I mean you would see the stack trace and where the error came from exactly. So maybe we could get rid of both try and catch in the `main` and the other function. It should give errors like this: 
![image](https://github.com/microsoft/bing-search-sdk-for-python/assets/36763164/36233407-8234-4167-af2d-a669426ef0cd)

Reply:
done. Thanks a lot for going the extra mile @kerolloz